rye/src/bootstrap.rs

```diff
@@ -55,6 +54,8 @@ virtualenv==20.25.0
 ruff==0.4.4
 "#;
 
+const LATEST_PIP: &str = "pip==23.3.2";
```

Comment:
We can drop it in another PR.

Reply:
I may have been mistaken but I thought we needed pip in the uv venv for tools like maturin

Reply:
Any other concerns @T-256 ?

Reply:
> I may have been mistaken but I thought we needed pip in the uv venv for tools like maturin

I don't think so.
uv is also drop-in replacement for pip, so I think we can use it instead.

I didn't do full review since a team member should do and approve this PR. I just recommended what could done in next PR.

Regard, @nazq .

Reply:
Seems you're correct. Looks like maturin added this support with this PR a couple of months back. https://github.com/PyO3/maturin/pull/2015